CRM/ManualDirectDebit/Hook/PostProcess/Contribution/DirectDebitMandate.php

```diff
@@ -16,10 +16,33 @@ public function __construct(&$form) {
     $this->form = $form;
   }
 
+  /**
+   * Checks if payment option appropriate for creating mandate
+   */
+  public function checkPaymentOptionToCreateMandate() {
+    if (isset($this->form->getVar('_params')['is_recur']) && !empty($this->form->getVar('_params')['is_recur'])){
+      $directDebitPaymentProcessor = civicrm_api3('PaymentProcessor', 'getvalue', array(
+        'return' => "id",
+        'name' => "direct debit",
+      ));
+      if($this->form->getVar('_params')['payment_processor_id'] == $directDebitPaymentProcessor){
+        $this->createMandate();
+      }
+    } else {
+      $directDebitPaymentInstrument = civicrm_api3('OptionValue', 'getvalue', array(
+        'return' => "value",
+        'name' => "direct_debit",
+      ));
+      if ($this->form->getVar('_params')['payment_instrument_id'] == $directDebitPaymentInstrument){
+        $this->createMandate();
+      }
+    }
+  }
```

Comment:
We try not to have really big if statements, simplifying them creating variables for each part of the if statement to improve readibility.

Example:
```php
$isRecurring = isset($this->form->getVar('_params')['is_recur']) && !empty($this->form->getVar('_params')['is_recur']);

if ($isRecurring) {
  $selectedPaymentProcessor = $this->form->getVar('_params')['payment_processor_id'];
  $directDebitPaymentProcessor = civicrm_api3('PaymentProcessor', 'getvalue', array(
    'return' => "id",
    'name' => "direct debit",
  ));

  if ($selectedPaymentProcessor == $directDebitPaymentProcessor) {
    $this->createMandate();
  }
} 
// This could be cool too...
else {
  if ($this->wasDirectDebitPaymentInstrumentSelected()){
    $this->createMandate();
  }
}
```

Either option would be good (creating variables or creating methods). The idea is to be able to know what the method is trying to do at a glance. ` isset($this->form->getVar('_params')['is_recur']) && !empty($this->form->getVar('_params')['is_recur']);` is hard to read. `$isRecurring` is obvious.